ui/test/nuxeo-delete-document-button.test.js

```diff
@@ -0,0 +1,215 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { fixture, flush, html, isElementVisible, waitForEvent } from '@nuxeo/nuxeo-elements/test/test-helpers.js';
+import { tap } from '@polymer/iron-test-helpers/mock-interactions.js';
+import { dom } from '@polymer/polymer/lib/legacy/polymer.dom';
+import '../actions/nuxeo-delete-document-button.js';
+/* eslint-disable no-unused-expressions */
+
+suite('<nuxeo-delete-document-button>', () => {
+  let button;
+
+  const deletableDocument = {
+    'entity-type': 'document',
+    uid: '1',
+    contextParameters: {
+      permissions: ['Remove'],
+    },
+    facets: [],
+    isTrashed: false,
+    isVersion: false,
+    type: 'File',
+  };
+
+  setup(async () => {
+    button = await fixture(
+      html`
+        <nuxeo-delete-document-button></nuxeo-delete-document-button>
+      `,
+    );
+  });
+
+  suite('Button Visibility', () => {
+    test('Should not be visible when document is null', async () => {
+      button.document = null;
+      await flush();
+      expect(isElementVisible(dom(button.root).querySelector('.action'))).to.be.false;
+    });
+
+    test('Should not be visible when document is a version', async () => {
+      button.document = {
+        isVersion: true,
+      };
+      await flush();
+      expect(isElementVisible(dom(button.root).querySelector('.action'))).to.be.false;
+    });
+
+    test('Should not be visible when no "Remove" permissions is granted', async () => {
+      button.document = {
+        contextParameters: {
+          permissions: ['Read'],
+        },
+        isVersion: false,
+      };
+      await flush();
+      expect(isElementVisible(dom(button.root).querySelector('.action'))).to.be.false;
+    });
+
+    test('Should not be visible when document is already trashed and no hard deletion is desired', async () => {
+      button.document = {
+        contextParameters: {
+          permissions: ['Remove'],
+        },
+        isTrashed: true,
+        isVersion: false,
+      };
+      await flush();
+      expect(isElementVisible(dom(button.root).querySelector('.action'))).to.be.false;
+    });
+
+    test('Should be visible otherwise', async () => {
+      button.document = deletableDocument;
+      await flush();
+
+      const actionDiv = dom(button.root).querySelector('.action');
+      expect(isElementVisible(actionDiv)).to.be.true;
+      expect(isElementVisible(actionDiv.querySelector('paper-icon-button'))).to.be.true;
+      expect(isElementVisible(actionDiv.querySelector('.label'))).to.be.false;
+    });
+
+    test('Should have corresponding information when "hard" property is set', async () => {
+      button.document = deletableDocument;
+      button.hard = true;
+      button.showLabel = true;
```

Comment:
This is more of a philosophical question: what's the goal of testing with `showLabel` set to `true`? Do we just want to check the label's value or do we want to check if it can show the label at all? If the second is true, then this is common to all actions, and it is code we'll be duplicating often. And if we're checking the showLabel, which is something that is only used on the actions menu, should we check for the tooltip as well? 🤔 

I'd say that, since this logic is common to all actions, we'd eventually test this once we extract it to some reusable class/mixin/behavior. Otherwise, we'll be repeatig

Reply:
It is something we need to discuss and think a little further. Testing visibility avoiding crossing other elements' borders or avoiding testing Polymer itself may, sometimes, be tricky.

Anyway, I'd say that we could extract it, yes. Even having a specific assert for that, that we can reuse. Or, indeed, extracting the label/tooltip logic to somewhere else. But, probably we should tackle it in our unit tests refactor task.